src/main/java/org/broadinstitute/hellbender/tools/walkers/ReferenceConfidenceVariantContextMerger.java

```diff
@@ -343,33 +345,51 @@ private static void removeStaleAttributesAfterMerge(final Map<String, Object> at
     /**
      * Adds attributes to the global map from the new context in a sophisticated manner
      *
-     * @param myAttributes               attributes to add from
+     * @param vcPair                     variant context with attributes to add from
      * @param annotationMap              map of annotations for combining later
      */
     @VisibleForTesting
-    static <T extends Comparable<? super T>> void addReferenceConfidenceAttributes(final Map<String, Object> myAttributes,
-                                                         final Map<String, List<Comparable>> annotationMap) {
-        for ( final Map.Entry<String, Object> p : myAttributes.entrySet() ) {
+    protected void addReferenceConfidenceAttributes(final VCWithNewAlleles vcPair, final Map<String, List<Object>> annotationMap) {
+        for ( final Map.Entry<String, Object> p : vcPair.getVc().getAttributes().entrySet() ) {
             final String key = p.getKey();
-            final Object value = p.getValue();
 
-            // add the annotation values to a list for combining later
-            List<Comparable> values = annotationMap.get(key);
-            if( values == null ) {
-                values = new ArrayList<>();
-                annotationMap.put(key, values);
-            }
-            try {
-                final String stringValue = value.toString();
-                // Branch to avoid unintentional, implicit type conversions that occur with the ? operator.
-                if (stringValue.contains(".")) {
-                    values.add(Double.parseDouble(stringValue));
-                } else {
-                    values.add(Integer.parseInt(stringValue));
+            // If the key corresponds to a requested reducible key, store the data as AlleleSpecificAnnotationData
+            if (annotatorEngine.isRequestedReducibleRawKey(key)) {
+                final List<Object> valueList = vcPair.getVc().getAttributeAsList(key);
+
+                List<ReducibleAnnotationData<?>> values = (List<ReducibleAnnotationData<?>>)(List<?>) annotationMap.get(key);
+                if (values == null) {
+                    values = new ArrayList<>();
+                    annotationMap.put(key, (List<Object>)(List<?>) values);
+                }
+                StringJoiner joiner = new StringJoiner(",");
+                for (Object s : valueList) joiner.add(s.toString());
+
+                ReducibleAnnotationData<Object> pairData = new AlleleSpecificAnnotationData<>(vcPair.getNewAlleles(), joiner.toString());
+                values.add(pairData);
```

Comment:
Its not needed here because values has already been added to the annotation map 4 lines above OR it originally came from the annotation map and its just being mutated in place.